internal/webhook/namespacemutation/webhookhandler.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package namespacemutation
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/pkg/instrumentation/auto"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-namespace,mutating=true,failurePolicy=ignore,groups="",resources=namespaces,verbs=create;update,versions=v1,name=mnamespace.kb.io,sideEffects=none,admissionReviewVersions=v1
+
+var _ admission.Handler = (*handler)(nil)
+
+type handler struct {
+	decoder            *admission.Decoder
+	annotationMutators *auto.AnnotationMutators
+}
+
+func NewWebhookHandler(decoder *admission.Decoder, annotationMutators *auto.AnnotationMutators) admission.Handler {
+	return &handler{
+		decoder:            decoder,
+		annotationMutators: annotationMutators,
+	}
+}
+
+func (h *handler) Handle(_ context.Context, req admission.Request) admission.Response {
+	namespace := &corev1.Namespace{}
+	err := h.decoder.Decode(req, namespace)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	h.annotationMutators.MutateObject(namespace)
+	marshaledNamespace, err := json.Marshal(namespace)
+	if err != nil {
+		res := admission.Errored(http.StatusInternalServerError, err)
+		res.Allowed = true
```

Comment:
Is there a specific reason why we are setting setting Allowed for the response there?

Reply:
Related Docs: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response

Docs specify that all responses must have an `allowed` setting. I guess in this case, the request is allowed but there was an error with marshalling the namespace.

Reply:
On a successful patch, `PatchResponseFromRaw` actually returns Allowed to true.

https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.0/pkg/webhook/admission/response.go#L88

Reply:
Interesting, the `Errored` func has Allowed to false which I presume we don't want.

Reply:
Yeah at this point it's a valid namespace, so even if we want to report an error, we still want to mark it as a valid request.